runtime/src/snapshot_utils.rs

```diff
@@ -5585,8 +5608,39 @@ mod tests {
         .unwrap();
 
         assert_eq!(
-            deserialized_bank, *bank,
+            deserialized_bank, bank,
             "Ensure rebuilding bank from the highest snapshot dir results in the highest bank",
         );
     }
+
+    #[test]
+    fn test_purge_old_bank_snapshots() {
+        solana_logger::setup();
+
+        let genesis_config = GenesisConfig::default();
+        let tmp_dir = tempfile::TempDir::new().unwrap();
+        let bank_snapshots_dir = tmp_dir.path();
+        let bank = create_snapshot_dirs_for_tests(&genesis_config, bank_snapshots_dir, 10, 5);
+        // Keep bank in this scope so that its account_paths tmp dirs are not released, and purge_old_bank_snapshots
+        // can clear the account hardlinks correctly.
+        let account_paths = &bank.rc.accounts.accounts_db.paths;
+        info!("account_paths: {:?}", account_paths);
```

Comment:
Originally, there was no "let bank =", so bank was not returned, so its tmpdir account_paths directories were removed.  Then remove_bank_snapshot fails because of the account_hardlinks io failures.

After adding ""let bank =", I thought I need to have some dummy way to use the bank variable so that its not being used won't cause any compilation error.  But after removing the line "let account_paths = &bank.rc.accounts.accounts_db.paths;"  there is no error message about bank not being used.
Not sure why.  So I have now removed that line.

Reply:
OK, I still have to use _bank to avoid the CI test failure.  But _bank does hold the tmpdir resource, so the test still passes.